File: python/pip/common_selenium/draganddrop_page.py
from pip.common_selenium.base_page import BasePage
from selenium.webdriver.remote.webdriver import WebDriver
from selenium.webdriver.common.by import *
from selenium.webdriver import ActionChains
from typing import Callable, Tuple, Any


class DragAndDropPage(BasePage):
    source: Tuple[str, str]
    target: Tuple[str, str]

    def __init__(self, driver: WebDriver):
        super().__init__(driver)
        self.source = (By.ID, "column-a")
        self.target = (By.ID, "column-b")

    def run_test(self, assert_provider: Callable[..., Any]):
        ActionChains(self.driver).drag_and_drop(
            self.get_present_element(self.source), self.get_present_element(self.target)
        ).perform()
        assert_provider(self.get_result(self.target) == "A")

    def get_result(self, selector):
        return self.get_present_element(selector).text

File: java/selenium-testng/src/test/java/com/browserstack/pages/DragAndDropPage.java
package com.browserstack.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;

public class DragAndDropPage extends BasePage {

    public DragAndDropPage(WebDriver driver) {
        super(driver);
    }

    public void performDragAndDrop() {
        WebElement source = driver.findElement(By.id("column-a"));
        WebElement target = driver.findElement(By.id("column-b"));
        Actions actions = new Actions(driver);
        actions.dragAndDrop(source, target).perform();
    }

    public String getResult() {
        WebElement target = driver.findElement(By.id("column-b"));
        return target.getText();
    }
}
